ddl/index.go

```diff
@@ -721,38 +713,77 @@ func onDropIndex(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64, _ error) {
 	return ver, errors.Trace(err)
 }
 
-func checkDropIndex(t *meta.Meta, job *model.Job) (*model.TableInfo, *model.IndexInfo, error) {
+func removeDependentHiddenColumns(tblInfo *model.TableInfo, idxInfo *model.IndexInfo) {
+	hiddenColOffs := make([]int, 0)
+	for _, indexColumn := range idxInfo.Columns {
+		col := tblInfo.Columns[indexColumn.Offset]
+		if col.Hidden {
+			hiddenColOffs = append(hiddenColOffs, col.Offset)
+		}
+	}
+	// Sort the offset in descending order.
+	sort.Slice(hiddenColOffs, func(i, j int) bool {
```

Comment:
```slices.SortFunc``` is better than ```sort.Slice```

1、it use pqqsort. it has better performance that quicksort of ````sort.Slice````
2、LessFunc avoid Bounds Check.